planner/core/optimizer.go

```diff
@@ -162,6 +162,7 @@ type logicalOptRule interface {
 func BuildLogicalPlanForTest(ctx context.Context, sctx sessionctx.Context, node ast.Node, infoSchema infoschema.InfoSchema) (Plan, types.NameSlice, error) {
 	sctx.GetSessionVars().PlanID.Store(0)
 	sctx.GetSessionVars().PlanColumnID.Store(0)
+	sctx.GetSessionVars().EnableHashJoin = true
```

Comment:
Too many CI failures due to the default value of this variable:
![image](https://github.com/pingcap/tidb/assets/7499936/b4aec0d0-cfc0-47ce-969a-f9ad0046e8f9)

Can we change the name of the variable from `EnableHashJoin` to `DisableHashJoin` and set it as `false` by default?

1) for most users, `HashJoin` should be allowed, so using `false` as the default value seems safer.
2) for some tests, the test framework won't use `DefEnableHashJoin` to initialize `sctx.EnableHashJoin`, instead it uses the default value of `Bool` in `Golang`, which is false; If we use `EnableHashJoin`, there will be plenty of test failures, changing it to `DisableHashJoin` can fix this.

Reply:
After discussing with PM, we decided to keep it unchanged as `EnableHashJoin`.

Reply:
> Too many CI failures... For safety, how about using another solution, which is using `DisableHashJoin` internally, and exposing `enable_hash_join` to users.

@coderplay I've pushed some code to your PR directly (tomorrow is the code freeze deadline of v7.4 ...) by using this approach to solve CI failures.

Reply:
Thanks for fixing those @qw4990 .  How can I detect those issues locally before submitting this PR?